src/hardware/configuration.rs

```diff
@@ -9,12 +9,12 @@ use stm32h7xx_hal::{
 
 use smoltcp_nal::smoltcp;
 
-use embedded_hal::digital::v2::{InputPin, OutputPin};
+pub use embedded_hal::digital::v2::{InputPin, OutputPin};
```

Comment:
Why make this a public use?

Reply:
Actually I wanted to import the trait from here and not from the hal. I'll fix that instead.

Cross-importing (in dual-iir) from the hal to bring a trait into scope for a type/struct that's defined somewhere else always annoys me. The usual `prelude` approach to get those traits also doesn't convince me. Maybe I'm doing it wrong.

Reply:
I tend to approach it as "import where necessary" - so I just import `embedded_hal::digital::v2::InputPin` wherever I need to use it.

Reply:
But that has two big disadvantages:

* Assume that the point of use is a different crate. Then you are needlessly creating a new dependency from the using crate to the trait crate. Re-exporting it would avoid that.
* It seems like a completely unrelated import. It's unclear from reading alone where (on what struct) that trait is actually used.

That seems to match https://github.com/rust-lang/api-guidelines/discussions/176 and could be addressed by the guidance described there.

Reply:
Since this is a bit bigger than just this single instance, I'll leave it like it is currently (re-export the trait) and leave the hole topic to be addressed en-bloc later.